### DISCUSSION TITLE
The right way to chain two event actions?

### DISCUSSION THREAD

**Discussion by dgvirtual** [alpinejs/alpine#3272] [1. Help] [answered]
I have a simple alpine.js based app which takes input from a form and generates a text on the fly. 

Then I have two buttons: one to highlight all the text that was added via input fields (this would add a class 'hgl' to the added pieces of text using Alpine.js functionality): 

````<button  x-on:click="hgl = ! hgl">Highlight</button>````

And the other, that employs a function external to alpine.js to copy the generated text. But I also need that button to unhighlight the text before the copy operation. The function to copy the text is perhaps not very important, but just in case, I include it at the end of this entry.

So anyway, the problem I had was that if I added both 'hgl = false' and the copy function (copyToClipboard) in one line: 

````<button x-on:click="hgl = false ; copyToClipboard('content')">Copy</button>````

, the copy action would happen before the Alpine.js removed the highlighting class from text, and the user would copy highlighted text. 

The way I solved it was by adding .debounce to the second action: 

````<button x-on:click="hgl = false" x-on:click.debounce="copyToClipboard('content')">Copy</button>````

I suspect, however, that postponing the second operation is not the correct way to approach the problem. What would be a better way? How do I chain the two functions when one is not internal to Alpine?

Here is my copy function (included in the header of the file):
````
const copyToClipboard = (elementId) => {
        const temp = document.createElement("div");
        temp.setAttribute("contentEditable", true);
        temp.innerHTML = document.getElementById(elementId).innerHTML;
        temp.setAttribute("onfocus", "document.execCommand('selectAll',false,null)");
        document.body.appendChild(temp);
        temp.focus();
        document.execCommand("copy");
        document.body.removeChild(temp);
    };
````

**Comment by SimoTod** [comment, marked as answer]
you need [nextTick](https://alpinejs.dev/magics/nextTick) to ensure Alpine updates the DOM when the state changes.

**Reply by dgvirtual** [reply]
outch, that was so simple (and I have read about it previously), but it just didn't cross my mind at the right time... Thank you @SimoTod
